index.js

```diff
@@ -2,328 +2,416 @@
+            return await certificateClient.getCertificate(certificateName);
+          } else {
+            return pollStatus();
+          }
+        });
+      }
+      return pollStatus();
+    })
+    .then((secret) => {
+      certificateSecretObj = secret;
+      print_item(secret);
+      return createNetwork();
+    })
+    .then((nic) => {
+      print_item(nic);
+      return createVirtualMachine(nic.id);
+    })
+    .then((vm) => {
+      print_item(vm);
+      return networkClient.publicIPAddresses.get(
+        resourceGroupName,
+        publicIPName
+      );
+    })
+    .then((ipAddress) => {
+      console.log(
+        `You can connect to the VM using: ssh ${adminUsername}@${ipAddress.ipAddress}`
+      );
+      console.log(`And password: ${adminPassword}\n`);
+      console.log(
+        "Your certificate is available in this folder: /var/lib/waagent"
+      );
+      console.log("You must be root to see it (sudo su)\n");
+      console.log(`/nDeleting the resource group: ${resourceGroupName}`);
+      // Comment the line below to retain your sample resources
+      return resourceClient.resourceGroups.beginDeleteAndWait(
+        resourceGroupName
+      );
+    })
+    .then((result) => {
+      console.log(`/nDeleted the resource group: ${resourceGroupName}`);
+    })
+    .catch((err) => {
+      throw err;
+    });
 }
 
-function createNetwork(){
+async function createNetwork() {
   let subnetObj;
-  return createVnet().then((vnet) => {
-    return networkClient.subnets.get(resourceGroupName, vnetName, subnetName);
-  }).then((subnet) => {
-    subnetObj = subnet;
-    return createPublicIP();
-  }).then((ip) => {
-    return createNIC(subnetObj, ip);
-  }).catch((err) => {
-    throw(err)}
-  );
+  return await createVnet()
+    .then(async (vnet) => {
+      return await networkClient.subnets.get(
+        resourceGroupName,
+        vnetName,
+        subnetName
+      );
+    })
+    .then(async (subnet) => {
+      subnetObj = subnet;
+      return await createPublicIP();
+    })
+    .then(async (ip) => {
+      return await createNIC(subnetObj, ip);
+    })
+    .catch((err) => {
+      throw err;
+    });
 }
 
-function createVnet() {
+async function createVnet() {
   let vnetParameters = {
     location: location,
     addressSpace: {
-      addressPrefixes: ['10.0.0.0/16']
+      addressPrefixes: ["10.0.0.0/16"],
     },
     dhcpOptions: {
-      dnsServers: ['10.1.1.1', '10.1.2.4']
+      dnsServers: ["10.1.1.1", "10.1.2.4"],
     },
-    subnets: [{ name: subnetName, addressPrefix: '10.0.0.0/24' }],
+    subnets: [{ name: subnetName, addressPrefix: "10.0.0.0/24" }],
   };
   console.log(`\n5.Creating vnet: ${vnetName} with subnet: ${subnetName}`);
-  return networkClient.virtualNetworks.createOrUpdate(resourceGroupName, vnetName, vnetParameters);
+  return await networkClient.virtualNetworks.beginCreateOrUpdateAndWait(
+    resourceGroupName,
+    vnetName,
+    vnetParameters
+  );
```

Comment:
we probably should call networkClient.virtualNetworks.get to get the full resourceinfo and return it back.